documentation/http_api_index.markdown

```diff
@@ -0,0 +1,141 @@
+
+### Authorization
+
+Authorization for `/puppet` endpoints is still controlled with Puppet's `auth.conf` authorization system.
+
+Puppet Server ignores `auth.conf` for `/puppet-ca` endpoints. Access to the `certificate_status` endpoint is configured in Puppet Server's `ca.conf` file, and the remaining CA endpoints are always accessible. Rack Puppet master servers still use `auth.conf` for `/puppet-ca`.
+
+When specifying authorization in `auth.conf`, the prefix, and the version number (such as `/puppet/v3`) on the paths must be retained, since Puppet matches authorization rules against the full request path.
+
+## Puppet V3 HTTP API
+
+The Puppet agent application uses several network services to manage systems. These services are all grouped under the `/puppet` API. Other tools can access these services and use the Puppet master's data for other purposes.
+
+The V3 API contains endpoints of two types: those that are based on dispatching to Puppet's internal "indirector" framework, and those that are not (namely the [environment endpoints](#environment-endpoints)).
+
+Every HTTP endpoint that dispatches to the indirector follows the form `/puppet/v3/:indirection/:key?environment=:environment`, where:
+
+-   `:environment` is the name of the environment that should be in effect for the request. Not all endpoints need an environment, but the query parameter must always be specified.
+-   `:indirection` is the indirection to which the request is dispatched.
+-   `:key` is the "key" portion of the indirection call.
+
+Using this API requires significant understanding of how Puppet's internal services are structured, but the following documents specify what is available and how to interact with it.
+
+### Configuration management services
+
+The Puppet agent application directly uses these servcies to manage the configuration of a node.
+
+These endpoints accept payload formats formatted as JSON or PSON (MIME types of `application/json` and `text/pson`, respectively) except for `File Content` and `File Bucket File`, which always use `application/octet-stream`.
+
+-   [Facts](https://puppet.com/docs/puppet/latest/http_api/http_facts.html)
+-   [Catalog](https://puppet.com/docs/puppet/latest/http_api/http_catalog.html)
+-   [Node](https://puppet.com/docs/puppet/latest/http_api/http_node.html)
+-   [File bucket file](https://puppet.com/docs/puppet/latest/http_api/http_file_bucket_file.html)
+-   [File content](https://puppet.com/docs/puppet/latest/http_api/http_file_content.html)
+-   [File metadata](https://puppet.com/docs/puppet/latest/http_api/http_file_metadata.html)
+-   [Report](https://puppet.com/docs/puppet/latest/http_api/http_report.html)
+
+### Informational services
+
+These services are not directly used by Puppet agent, but can be used by other tools.
+
+-   [Status](https://puppet.com/docs/puppet/latest/http_status.html)
```

Comment:
This exists and should be documented but folks should prefer the Puppet Server status api instead: https://github.com/puppetlabs/puppetserver/tree/master/documentation/status-api/v1

Also this link is broken (it needs an `http_api` in it)